src/solana/SolanaEngine.ts

```diff
@@ -273,52 +278,47 @@ export class SolanaEngine extends CurrencyEngine<
       return
     }
 
-    let failedTxQueryIndex = -1
+    const transactionRequests: RpcRequest[] = []
     for (let i = txids.length - 1; i >= 0; i--) {
-      // Process the transactions from oldest to newest
-      try {
-        const tx = asRpcGetTransaction(
-          await this.fetchRpc('getTransaction', [
-            txids[i].signature,
-            { encoding: 'json', commitment: this.networkInfo.commitment }
-          ])
-        )
+      transactionRequests.push({
+        method: 'getTransaction',
+        params: [
+          txids[i].signature,
+          { encoding: 'json', commitment: this.networkInfo.commitment }
+        ]
+      })
+    }
+    const txResponse: RpcGetTransaction[] = await this.fetchRpcBulk(
+      transactionRequests
+    )
+    const slots = txResponse.map(res => asTransaction(res).result.slot)
+    const blocktimeRequests: RpcRequest[] = slots.map(slot => ({
+      method: 'getBlockTime',
+      params: [slot]
+    }))
+    const blocktimeResponse: Blocktime[] = await this.fetchRpcBulk(
+      blocktimeRequests
+    )
 
-        // From testing, getSignaturesForAddress always returns a blocktime but it is optional in the RPC docs so we should be prepared to get it if it isn't present
-        let timestamp = txids[i].blocktime
-        if (timestamp == null || isNaN(timestamp))
-          timestamp = asNumber(await this.fetchRpc('getBlockTime', [tx.slot]))
-
-        this.processSolanaTransaction(tx, timestamp)
-
-        // Update progress
-        const percent = 1 - i / txids.length
-        if (percent !== this.progressRatio) {
-          if (Math.abs(percent - this.progressRatio) > 0.25 || percent === 1) {
-            this.progressRatio = percent
-            this.tokenCheckTransactionsStatus[this.chainCode] =
-              this.progressRatio
-            this.updateOnAddressesChecked()
-          }
+    // Process the transactions from oldest to newest
+    for (let i = txids.length - 1; i >= 0; i--) {
+      this.processSolanaTransaction(
+        txResponse[i].result,
+        asBlocktime(blocktimeResponse[i]).result
+      )
+      this.otherData.newestTxid = txids[i].signature
+
+      // Update progress
+      const percent = 1 - i / txids.length
+      if (percent !== this.progressRatio) {
+        if (Math.abs(percent - this.progressRatio) > 0.25 || percent === 1) {
+          this.progressRatio = percent
+          this.tokenCheckTransactionsStatus[this.chainCode] = this.progressRatio
+          this.updateOnAddressesChecked()
```

Comment:
It seems weird to maintain a fractional progress ratio in here. Since it's only one network query now, wouldn't we just report 0 (not queried) or 1 (queried) for each token?

Even if there is some paging going on, I would expect the progress code to move outside the loop, at the same level as the network request, instead of inside the result-processing loop.